File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include<stack>

using namespace std;

stack<char> stk;

bool solution(string s)
{
    bool answer = true;
    
    int size=s.length();
    for(int i=0;i<size;i++){
        if(s[i]=='('){
            stk.push('(');
        }
        else{
            if(!stk.empty()){
                stk.pop();
            }
            else{
                return false;
            }
        }
    }
    if(!stk.empty())
        answer=false;
    

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = false;
        Stack<Character> stk = new Stack<>();
        
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                stk.push(s.charAt(i));
            }
            else{
                if(stk.isEmpty()){
                    return false;
                }
                else{
                    stk.pop();
                }
            }
        }
        
        if(stk.isEmpty()){
            answer=true;
        }
        
        return answer;
    }
}